api: zbus: Make zbus a stable API

rodrigopex opened the pull request:
Make zbus stable API since it fulfills all the requirements that makes an API stable.

rodrigopex committed:
api: zbus: Make zbus a stable API

Make zbus stable API since it fulfills all the requirements that
makes an API stable.

Signed-off-by: Rodrigo Peixoto <rodrigopex@gmail.com>

carlescufi commented on include/zephyr/zbus/zbus.h:
```diff
@@ -18,6 +18,8 @@ extern "C" {
 /**
  * @brief Zbus API
  * @defgroup zbus_apis Zbus APIs
+ * @since 4.2.0
```
Not sure if this should instead be the version of the introduction of the API vs the version of its first versioning. @kartben?

rodrigopex replied:
Zbus was introduced in Zephyr v3.3.0. This "since" was not clear to me.

carlescufi replied:
Let's discuss it in the Architecture WG meeting.

rodrigopex replied:
Done.

carlescufi commented:
Architecture WG meeting:

- No objections but #88834 must be merged before.

rodrigopex committed:
doc: releases: State zbus API as stable

Zbus has been around since Zephyr v3.3.0 and fulfills all the requirements
to be a stable API.

Signed-off-by: Rodrigo Peixoto <rodrigopex@gmail.com>
